frontend/frontend.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# Install dependencies
+apt-get update && apt-get install -y libglib2.0-dev libfdt-dev libpixman-1-dev zlib1g-dev ninja-build libcap-ng-dev libslirp-dev
+
+
+# Install qemu
+if dpkg -s qemu >/dev/null 2>&1
+then
+    wget https://download.qemu.org/qemu-8.0.0.tar.xz
+    tar xvJf qemu-8.0.0.tar.xz
+    cd qemu-8.0.0
+    ./configure --prefix=/usr --enable-debug --target-list=x86_64-softmmu --enable-kvm --enable-virtfs --enable-slirp
+    make -j$(nproc)
+    make install
+    cd ..
+else
+    echo "qemu is installed"
+fi
+
+
+
+
+
+# Download VM image
+
+if [ -f "./focal-server-cloudimg-amd64.img" ]
+then
+    echo "File exists"
+else
+    wget https://cloud-images.ubuntu.com/focal/20230420/focal-server-cloudimg-amd64.img
+fi
+
+
+# Generate password hash
+password=$(openssl passwd -1 123456)
```

Comment:
leaking passwords is risky.

Reply:
@yishangzhang It is reasonable to delegate the action of setting password to the user.